lib/response/headers_builder.ex

```diff
@@ -32,6 +32,24 @@ defmodule HTTPServer.Response.HeadersBuilder do
     |> content_type("text/css;charset=utf-8")
   end
 
+  def content(headers, :jpeg, body) do
+    headers
+    |> content_length(body)
+    |> content_type("image/jpeg")
+  end
+
+  def content(headers, :png, body) do
+    headers
+    |> content_length(body)
+    |> content_type("image/png")
+  end
+
+  def content(headers, :gif, body) do
+    headers
+    |> content_length(body)
+    |> content_type("image/gif")
+  end
```

Comment:
I wonder if you could refactor this to use a `content_type` map? similar to how you handle ]HTTP status codes](https://github.com/kelseyroy/http_server/blob/b26ac34eb91efb3aae88a231929c0ab1633eef41/lib/response.ex#L54-L62)?

```Elixir
defp content_type(media_type) do
  %{
    :text => "text/plain;charset=utf-8",
    :html => "text/html;charset=utf-8",
    :json => "application/json;charset=utf-8",
    :xml => "application/xml;charset=utf-8",
    :css => "text/css;charset=utf-8",
    :jpeg => image/jpeg",
    :png => "image/png",
    :gif => "image/gif"
  }[media_type]
end

defp content_type(headers, media_type), do: %{headers | content_type: content_type(media_type)}

def content(headers, media_type, body) do
  headers
  |> content_length(body)
  |> content_type(media_type)
end
```

Reply:
Loved this idea!!! Just updated HeadersBuilder and it looks far easier to read/expand upon.